pages/docs/tooling/tooling-aste.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Artificial Solver Testing Environment (ASTE)
+permalink: tooling-aste.html
+keywords: tooling, aste
+summary: "A guide to usage of artifical solver testing environment (ASTE) tool"
+---
+
+## Motivation
+
+ASTE is a tool which can be used to understand the dynamics of preCICE. It is a lightweight wrapper around preCICE which allows to investigate different mapping setups in terms of both runtime and accuracy. It can be replace individual participant for development purpouses. It consist of C++ and Python modules.
+
+## Installation
+
+ASTE relies on following requirements
+
+### C++ modules
+
+#### Required dependencies for C++
+
+- C++ compiler (with support for C++14, e.g. GCC version >= 5)
+- preCICE
+- MPI
+- CMake (version >= 3.1)
+- Boost (version >= 1.65.1)
+- vtk (Visualization ToolKit)
+
+##### Optional dependencies for C++
+
+- METIS
+
+METIS is a graph partitioning library, it is used for topological partitioning in mesh partitioner. 
+
+### Python modules
+
+#### Required packages for Python
+
+- NumPy
+- vtk (Visualization Toolkit)
+
+Can be installed easily using pip
+
+```bash
+pip3 install numpy vtk
+```
+
+### Building and Installation
+
+In order to install ASTE on your system you can follow these steps:
+
+```bash
+mkdir build
+cd build
+cmake ..
+make
+sudo make install
+```
+
+## Usage
+
+There are some flags are common in every module here is a list of them.
+
+| Flag     | Explanation        |
+| -------- | ------------------ |
+| --mesh   | Mesh prefix        |
+| --data   | Name of data array |
+| --output | Output filename    |
+
+For demo usage please check [ASTE Tutorial](https://github.com/precice/tutorials/tree/develop).
+
+### preciceMap
+
+The main tool used for mapping. It is a wrapper around preCICE interface. It can be run in serial or parallel. Mapped data can be a vector or scalar.
+
+| Flag      | Explanation                                                   |
+| --------- | ------------------------------------------------------------- |
+| -v        | Enables verbose output                                        |
+| -c        | To specify preCICE config file (default="precice-config.xml") |
+| -p        | Participant name (A or B)                                     |
+| --runName | Name of run (default="")                                      |
+| --vector  | A bool switch to specify vector data (default=False)          |
+
+For example, mapping the data "x + y" from a mesh named fine_mesh in directory fine_mesh to anotherMesh and save into "mappedData" and "mappedMesh":
+
+```bash
+preciceMap -v -p A --mesh fine_mesh/fine_mesh --data "x + y" 
+mpirun -n 4 -v -p B --mesh anotherMesh --data "mappedData" --output mappedMesh
```

Comment:
We should also add a note that the `--mesh` argument here is different from the mesh name in the precice configuration file. Maybe refer to the tutorial xml file for references.